libflux/src/core/parser/mod.rs

```diff
@@ -577,6 +577,76 @@ impl Parser {
             MonoType::Invalid
         }
     }
+
+    // Record = "{" [ Identifier (Suffix1 | Suffix2) ] "}"
+    // Suffix1 = ":" MonoType { "," Property }
+    // Suffix2 = "with" [Properties]
+    #[cfg(test)]
+    fn parse_record(&mut self) -> Record {
+        let start = self.open(TOK_LBRACE, TOK_RBRACE);
+        let mut properties: Option<Vec<PropertyType>> = None;
+        let property: PropertyType;
+
+        let mut id: Option<Identifier> = None;
+        let _id;
```

Comment:
No need for this forward declaration either.

Reply:
I could be wrong about this one, how would I reference it in the else block when it goes out of scope?

Reply:
You assign it in the outermost if block, so it's still visible from inside the inner if/else blocks.

Reply:
oh very true.